baselines/fjord/fjord/main.py

```diff
@@ -0,0 +1,288 @@
+    model_path = hydra.core.hydra_config.HydraConfig.get(
+        )['runtime']['output_dir']
+
+    Logger.get().info(
+        f"Training on {device} using PyTorch "
+        f"{torch.__version__} and Flower {fl.__version__}"
+    )
+
+    trainloader, testloader = load_data(path, cid=0, seed=args.manual_seed,
+                                        train_bs=args.batch_size)
+    NUM_CLIENTS = args.num_clients
+    if args.client_tier_allocation == 'uniform':
+        cid_to_max_p = {cid: (cid // 20) * 0.2 + 0.2
+                        for cid in range(100)}
+    else:
+        raise ValueError(f"Client to tier allocation strategy "
+                         f"{args.client_tier_allocation} not currently"
+                         "supported")
+
+    model = get_net(args.model, args.p_s, device=device)
+    config = get_agg_config(model, trainloader, args.p_s)
+    train_config = SimpleNamespace(**{
+        "batch_size": args.batch_size,
+        "lr": args.lr,
+        "optimiser": args.optimiser,
+        "momentum": args.momentum,
+        "nesterov": args.nesterov,
+        "lr_scheduler": args.lr_scheduler,
+        "weight_decay": args.weight_decay,
+        "local_epochs": args.local_epochs,
+    })
+
+    if args.strategy == "fjord_fedavg":
+        strategy = FjORDFedAVG(
+            fraction_fit=args.sampled_clients/args.num_clients,
+            fraction_evaluate=0.,
+            min_fit_clients=args.min_fit_clients,
+            min_evaluate_clients=1,
+            min_available_clients=NUM_CLIENTS,
+            evaluate_fn=get_eval_fn(args, model_path, testloader, device),
+            on_fit_config_fn=get_fit_config_fn(args.num_rounds,
+                                               args.lr),
+            initial_parameters=fl.common.ndarrays_to_parameters(
+                get_parameters(get_net(args.model, args.p_s, device=device)))
+        )
+    else:
+        raise ValueError(f"Strategy {args.strategy} "
+                         "is not currently supported")
+
+    client_resources = args.client_resources
+    if device.type != "cuda":
+        client_resources = {"num_cpus": args.client_resources["num_cpus"],
+                            "num_gpus": 0}
+
+    if args.client_selection == 'balanced':
+        cl_manager = FjORDBalancedClientManager(cid_to_max_p)
+    elif args.client_selection == 'random':
+        cl_manager = None
+    else:
+        raise ValueError(
+            f"Client selection {args.client_selection}"
+            " is not currently supported")
+
+    Logger.get().info("Starting simulated run.")
+    # Start simulation
+    fl.simulation.start_simulation(
+        client_fn=get_client_fn(args, model_path, cid_to_max_p,
+                                config, train_config, device),
+        num_clients=NUM_CLIENTS,
+        config=fl.server.ServerConfig(num_rounds=args.num_rounds),
+        strategy=strategy,
+        client_resources=client_resources,
```

Comment:
Let's pass this directly without parsing, it could be that the majority of the work (i.e. client side of things) is done on a different node than where the server lives.